irr/include/irrString.h

```diff
@@ -176,10 +176,11 @@ class string
 		// no longer allowed!
 		_IRR_DEBUG_BREAK_IF((void *)c == (void *)c_str());
```

Comment:
I would rather extend this condition to entirely disallow assing a string to itself